### PR TITLE
TW-562 web improvements of input row

### PR DESCRIPTION
**Pull request by imGok**
#562

Demo :
webimrpvo

design : https://www.figma.com/file/uLw3iWnOEGaRlJnfPz4Pv0/Twake-%5BMatrix%5D?node-id=11116%3A182605&mode=dev
image
image

dmepobioebieb

### CODE REVIEW

**Comment by nqhhdev** [comment]
I think `send icon` will also be below like the `plus icon`

File: lib/pages/chat/chat_input_row.dart [hunk 1]
```diff
@@ -214,6 +118,176 @@ class ChatInputRow extends StatelessWidget {
       ),
     );
   }
+
+  Container _buildMobileInputRow(BuildContext context) {
+    return Container(
+      alignment: Alignment.center,
+      padding: const EdgeInsets.only(left: 12.0),
+      margin: const EdgeInsets.only(right: 8.0),
```
Comment (nqhhdev) [inline lib/pages/chat/chat_input_row.dart:126]: move it to `chat_input_row_style`

File: lib/pages/chat/chat_input_row.dart [hunk 2]
```diff
@@ -214,6 +118,176 @@ class ChatInputRow extends StatelessWidget {
...
+      decoration: BoxDecoration(
+        borderRadius: const BorderRadius.all(Radius.circular(25)),
```
Comment (nqhhdev) [inline lib/pages/chat/chat_input_row.dart:128]: same

File: lib/pages/chat/chat_input_row.dart [hunk 3]
```diff
@@ -214,6 +118,176 @@ class ChatInputRow extends StatelessWidget {
...
+  }
+
+  Container _buildWebInputRow(BuildContext context) {
+    return Container(
+      alignment: Alignment.center,
+      margin: const EdgeInsets.only(right: 8.0),
+      decoration: BoxDecoration(
+        borderRadius: const BorderRadius.all(Radius.circular(25)),
+        color: Theme.of(context).colorScheme.surface,
+      ),
```
Comment (nqhhdev) [inline lib/pages/chat/chat_input_row.dart:195]: same

**Comment by hoangdat** [comment]
image

IMO, the `send` button also need in bottom.

File: lib/pages/chat/chat_input_row.dart [hunk 4]
```diff
@@ -95,101 +89,9 @@ class ChatInputRow extends StatelessWidget {
...
-                                        );
-                                      },
-                                    ),
-                            ),
-                          ),
-                        ),
-                      ],
-                    ),
-                  ),
+                  child: PlatformInfos.isWeb
```
Comment (hoangdat) [inline lib/pages/chat/chat_input_row.dart:92]: Why not use the `responsiveUtil`? It also the same in `desktop`.

File: lib/pages/chat/chat_input_row.dart [hunk 5]
```diff
@@ -203,7 +105,9 @@ class ChatInputRow extends StatelessWidget {
                     height: 56,
                     alignment: Alignment.center,
                     child: TwakeIconButton(
-                      size: ChatInputRowStyle.sendIconButtonSize,
+                      size: PlatformInfos.isMobile
```
Comment (drminh2807) [inline lib/pages/chat/chat_input_row.dart:108]: move to style file

**Review by drminh2807 — CHANGES_REQUESTED** [review]

**Review by nqhhdev — APPROVED** [review]
Look good to me

File: lib/pages/chat/chat_input_row.dart [hunk 6]
```diff
@@ -214,6 +127,178 @@ class ChatInputRow extends StatelessWidget {
       ),
     );
   }
+
+  Container _buildMobileInputRow(BuildContext context) {
```
Comment (sherlockvn) [inline lib/pages/chat/chat_input_row.dart:131]: should be another stateless widget

File: lib/pages/chat/chat_input_row.dart [hunk 7]
```diff
@@ -214,6 +127,178 @@ class ChatInputRow extends StatelessWidget {
...
+                      ),
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Container _buildWebInputRow(BuildContext context) {
```
Comment (sherlockvn) [inline lib/pages/chat/chat_input_row.dart:199]: sanme

File: lib/pages/chat/chat_input_row.dart [hunk 8]
```diff
@@ -214,6 +127,178 @@ class ChatInputRow extends StatelessWidget {
...
+                      ),
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  InputBar _buildInputBar(BuildContext context) {
```
Comment (sherlockvn) [inline lib/pages/chat/chat_input_row.dart:277]: same

File: lib/pages/chat/chat_input_row_style.dart [hunk 9]
```diff
@@ -1,5 +1,35 @@
...
+import 'package:fluffychat/utils/responsive/responsive_utils.dart';
+import 'package:flutter/material.dart';
+
 class ChatInputRowStyle {
+  static final ResponsiveUtils responsiveUtils = getIt.get<ResponsiveUtils>();
+
   static double get chatInputRowWidth => 52.0;
-  static double get chatInputRowHeight => 56.0;
-  static double get sendIconButtonSize => 34.0;
+  static double get chatInputRowHeightWeb => 56.0;
```
Comment (drminh2807) [inline lib/pages/chat/chat_input_row_style.dart:9]:
```suggestion
  static const double chatInputRowHeightWeb = 56.0;
```
using const to improve render time

File: lib/pages/chat/chat_input_row.dart [hunk 10]
```diff
@@ -80,116 +80,30 @@ class ChatInputRow extends StatelessWidget {
...
             : <Widget>[
-                TwakeIconButton(
-                  tooltip: L10n.of(context)!.more,
-                  margin: const EdgeInsets.only(right: 4.0),
-                  icon: Icons.add_circle_outline,
-                  onTap: () => controller.onSendFileClick(context),
-                ),
+                if (ChatInputRowStyle.responsiveUtils.isMobileOrTablet(context))
+                  Container(
+                    margin: const EdgeInsets.only(right: 4.0),
```
Comment (drminh2807) [inline lib/pages/chat/chat_input_row.dart:85]: move to style file

File: lib/pages/chat/chat_input_row.dart [hunk 11]
```diff
@@ -80,116 +80,30 @@ class ChatInputRow extends StatelessWidget {
...
+                    height: ChatInputRowStyle.chatInputRowHeightMobile,
+                    child: TwakeIconButton(
+                      size: 24.0,
```
Comment (drminh2807) [inline lib/pages/chat/chat_input_row.dart:88]: same

File: lib/pages/chat/chat_input_row.dart (hunk 8 above)
Comment (imGok) [reply lib/pages/chat/chat_input_row.dart:288]: I agree for the other 2 but why this one ?

Comment (sherlockvn) [reply lib/pages/chat/chat_input_row.dart:154]: because it's also a stateless widget, so make it as much as possible, it will be easier when you debug.

Comment (imGok) [reply lib/pages/chat/chat_input_row.dart:154]: @nqhhdev @drminh2807 @hoangdat wdyt ? for me we'll create a widget with same arguments there is no big performance gain

Comment (nqhhdev) [reply lib/pages/chat/chat_input_row.dart:154]: IMO: No need create a stl

Comment (hoangdat) [reply lib/pages/chat/chat_input_row.dart:154]: IMO, no need this case

**Review by hoangdat — APPROVED** [review]

**Review by drminh2807 — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

**Pull request merged**